sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/DelegatingPerWorkerHistogram.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.metrics;
+
+import java.io.Serializable;
+import org.apache.beam.sdk.annotations.Internal;
+import org.apache.beam.sdk.util.HistogramData;
+
+/** Implementation of {@link Histogram} that delegates to the instance for the current context. */
+@Internal
+public class DelegatingPerWorkerHistogram implements Metric, Histogram, Serializable {
+  private final MetricName name;
+  private final HistogramData.BucketType bucketType;
+  private final boolean processWideContainer;
+
+  public DelegatingPerWorkerHistogram(
+      MetricName name, HistogramData.BucketType bucketType, boolean processWideContainer) {
+    this.name = name;
+    this.bucketType = bucketType;
+    this.processWideContainer = processWideContainer;
+  }
+
+  public DelegatingPerWorkerHistogram(MetricName name, HistogramData.BucketType bucketType) {
+    this(name, bucketType, false);
+  }
+
+  @Override
+  public void update(double value) {
+    MetricsContainer container =
+        processWideContainer
+            ? MetricsEnvironment.getProcessWideContainer()
+            : MetricsEnvironment.getCurrentContainer();
+    if (container != null) {
+      container.getPerWorkerHistogram(name, bucketType).update(value);
```

Comment:
same idea here

Reply:
Extended DelegatingHistogram to also handle perWorkerHistogram